tools/translations/process_source_ts_files.py

```
@@ -108,67 +113,106 @@ def write_ts_file(target_file, root):
+                    'Use the ellipsis character (…) instead.')
+
+            if "'" in tr_txt:
+                tr_error("Translatable string contains the straight single quote mark (').",
+                    'Use left (‘) or right (’) curly single quote mark, or prime (′).')
+
+            if '"' in re.sub(r'<a href="[^"]*">', '', tr_txt):
+                tr_error('Translatable string contains the straight double quote mark (").',
+                    'Use left (“) or right (”) curly double quote mark, or double prime (″).')
+
+            if plurals:
+                for idx, plural in enumerate(plurals):
```

Ruff (0.16.1)

[warning] Loop control variable `idx` not used within loop body

(B007)